src/test/java/picard/vcf/SamTestUtils.java

```diff
@@ -30,8 +29,17 @@ public static File createIndexedBamOrCram(final File samFile, final File tempFil
 
         final File output = File.createTempFile(tempFilePrefix.getAbsolutePath(), "." + type.fileExtension());
         output.deleteOnExit();
-        final File indexFile = new File(output.getAbsolutePath() + "." + type.indexExtension());
-        indexFile.deleteOnExit();
+        final File indexFile; //SamReaderFactor uses diffefent conventions for naming Bam indexes and Cram indexes, need to handle each case separately
+        if (type == SamReader.Type.BAM_TYPE) {
+            final String fileBase = output.getAbsolutePath().substring(0, output.getAbsolutePath().lastIndexOf('.'));
+            indexFile = new File(fileBase + BAMIndex.BAMIndexSuffix);
```

Comment:
This constant (`BAMIndex.BAMIndexSuffix`) has been deprecated in htsjdk. Probably best to use the replacement `BAMIndex#BAI_INDEX_SUFFIX`.

Reply:
done